presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -381,6 +382,48 @@ public void testArrays()
                                 1, ImmutableMap.of(new Subfield("c[1]"), IS_NULL))));
     }
 
+    @Test
+    public void testArraysWithSubfieldPruning()
+            throws Exception
+    {
+        tester.assertRoundTripWithSettings(arrayType(INTEGER),
+                createList(NUM_ROWS, i -> ImmutableList.of(1, 2, 3, 4)),
+                ImmutableList.of(
+                        OrcReaderSettings.builder().addRequiredSubfields(0, "c[1]").build(),
+                        OrcReaderSettings.builder().addRequiredSubfields(0, "c[1]", "c[2]").build(),
+                        OrcReaderSettings.builder().addRequiredSubfields(0, "c[2]").build()));
+
+        Random random = new Random(0);
```

Comment:
We rarely use random in unit tests. Can we make a fixed list?

Reply:
I've been using it a lot here. I don't see a simple way to change that right now. My understanding is that using random is problematic because it makes tests harder to reproduce, however, I think using random with constant seed mitigates that concern. Is there any other reason to avoid random?